src/index.js

```diff
@@ -177,7 +192,9 @@ class DropdownTreeSelect extends Component {
       document.removeEventListener('click', this.handleOutsideClick, false)
     }
 
-    this.setState(nextState)
+    this.setState(nextState, () => {
+      callback && callback(tags)
```

Comment:
This would need `typeof callback === 'function' && callback(tags)` if you truly wanna be defensive but in general, I think we should see if we can eliminate the need to rely on `setState` callbacks. We'll probably have to refactor existing event handlers in a way that the setState portion can be extracted out into a reusable block - but I'm not sure if the resulting code would be better than having the callbacks. I'll take a deeper dive tomorrow but just leaving this note out here in case you have any ideas about it.

Reply:
Nope, no other ideas, have to wait for state to be set. This is the fastest way there.